### DISCUSSION TITLE
Can the “noNetScaleSigma” option be added to the “staticPositions” for parameterization in the “GnssProcessing” program?

### DISCUSSION THREAD

**Discussion by IamLittleMonster** [groops-devs/groops#226] [Ideas]
Dear developer,
In the scenario of “**GnssNetwork**”, the file of "**030groopsGnssProcessing.xm**l” shows the example about the program “**GnssProcessing**”.For the "**staticPositions parameterization**", there are currently only two options available: noNetTranslation and noNetRotation. Can we **add noNetScaleSigma to correspond to noNetScale?**

For the above issues, I have found possible modifications in lines 233 to 247 of/source/gnss/gnssParameterization/gnssParameterizationStaticPositions.cpp/GnssParametrizationStaticPositions::constraints. The specific code is as follows:
    for(UInt idRecv=0; idRecv<gnss->receivers.size(); idRecv++)
      if(selectedNoNetReceivers.at(idRecv) && index.at(idRecv))
      {
        copy((noNetPos.at(idRecv)-pos.at(idRecv)).vector(), l.row(3*i, 3));
        if(sigmaNoNetTranslation)
          copy(identityMatrix(3), A.slice(3*i, idxNNT, 3, 3));
        if(sigmaNoNetRotation)
        {
          const Vector3d pos = noNetPos.at(idRecv)/DEFAULT_R;
          A(3*i+0, idxNNR+1) =  pos.z(); A(3*i+0, idxNNR+2) = -pos.y();
          A(3*i+1, idxNNR+0) = -pos.z(); A(3*i+1, idxNNR+2) =  pos.x();
          A(3*i+2, idxNNR+0) =  pos.y(); A(3*i+2, idxNNR+1) = -pos.x();
        }
        i++;
      }

In my view, developers only need to emulate the above code and add corresponding conditions.
However, because I don't have experience in developing Qt desktop applications, I don't know how to modify the GUI version of Groops.
I think this is a very useful improvement, and I hope developers can consider it in their next plans.

Best Regards,
Hongzhan

**Comment by tmayerguerr** [comment]
Hi Hongzhan,
see in my [answer](https://github.com/groops-devs/groops/discussions/221#discussioncomment-7474574) in #221.
regards
Torsten

**Reply by IamLittleMonster** [reply]
Hello Torsten,
Thanks for your reply. Next, I will try steps which you have mentioned above.

Regards,
Hongzhan
